src/System.Linq.Expressions/tests/IndexExpression/IndexExpressionTests.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections.Generic;
+using System.Reflection;
+using Xunit;
+
+namespace System.Linq.Expressions.Tests.IndexExpression
+{
+    public static class IndexExpressionTests
+    {
+        [Fact]
+        public static void UpdateSameTest()
+        {
+            var instance = new SampleClassWithProperties { DefaultProperty = new List<int> { 100, 101 } };
+            Expressions.IndexExpression expr = instance.DefaultIndexExpression;
+
+            var exprUpdated = expr.Update(expr.Object, expr.Arguments);
+
+            // Has to be the same, because everything is the same.
+            Assert.Same(expr, exprUpdated);
+
+            // Invoke to check expression.
+            IndexExpressionHelpers.AssertInvokeCorrect(100, expr, instance);
+            IndexExpressionHelpers.AssertInvokeCorrect(100, exprUpdated, instance);
+        }
+
+        [Fact]
+        public static void UpdateTest()
+        {
+            var instance = new SampleClassWithProperties
+            {
+                DefaultProperty = new List<int> { 100, 101 },
+                AlternativeProperty = new List<int> { 200, 201 }
+            };
+
+            var expr = instance.DefaultIndexExpression;
+            var newProperty = Expression.Property(Expression.Constant(instance),
```

Comment:
I'll stop commenting on the vars, but in general, please use the type name unless the type is explicitly mentioned on the right-hand side, e.g. a cast, a ctor, etc.